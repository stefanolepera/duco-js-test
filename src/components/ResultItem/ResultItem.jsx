import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Title from './components/Title/Title';
import SubTitle from './components/SubTitle/SubTitle';
import { formatDetail } from '../../utils/FilterData';

const MainWrapper = styled.div`
    margin: 0 auto;
    width: 620px;
    min-width: 400px;
    background: #ccc;
    border-radius: 4px;
`;

const ContentWrapper = styled.div`
    display: flex;
    align-items: flex-start;
    justify-content: space-around;
`;

const DetailsWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: flex-start;
    margin: 0 20px;
`;

const ResultItem = ({ characterName, characterDetails, characterFilms }) => (
    <MainWrapper>
        <Title title={characterName} />
        <hr />
        <ContentWrapper>
            <DetailsWrapper>
                <SubTitle subTitle="Details" />
                {characterDetails.map((detail, index) => (
                    <div key={index}>
                        {formatDetail(detail[0])} : {detail[1]}
                    </div>
                ))}
                <hr />
            </DetailsWrapper>
            <DetailsWrapper>
                <SubTitle subTitle="Films" />
                {characterFilms.map((film, index) => (
                    <div key={index}>{film}</div>
                ))}
            </DetailsWrapper>
        </ContentWrapper>
    </MainWrapper>
);

ResultItem.propTypes = {
    characterName: PropTypes.string,
    characterDetails: PropTypes.arrayOf(PropTypes.array),
    characterFilms: PropTypes.arrayOf(PropTypes.string)
};

export default ResultItem;
