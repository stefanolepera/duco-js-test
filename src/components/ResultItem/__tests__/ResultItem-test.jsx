import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ResultItem from '../ResultItem';

describe('<ResultItem /> test', () => {
    it('Test snapshots of content', () => {
        const wrapper = shallow(
            <ResultItem
                characterName={''}
                characterDetails={[]}
                characterFilms={[]}
            />
        );

        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });
});
