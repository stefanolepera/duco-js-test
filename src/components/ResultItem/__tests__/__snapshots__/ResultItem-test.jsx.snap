// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultItem /> test Test snapshots of content 1`] = `
<styled.div>
  <Title
    title=""
  />
  <hr />
  <styled.div>
    <styled.div>
      <SubTitle
        subTitle="Details"
      />
      <hr />
    </styled.div>
    <styled.div>
      <SubTitle
        subTitle="Films"
      />
    </styled.div>
  </styled.div>
</styled.div>
`;
